lib/asset/AssetModulesPlugin.js

```diff
@@ -207,7 +208,12 @@ class AssetModulesPlugin {
 									render: () =>
 										/** @type {Source} */ (codeGenResult.sources.get(type)),
 									filename: errored
-										? module.nameForCondition()
+										? path.isAbsolute(module.nameForCondition())
+											? path.relative(
+													compiler.context,
+													module.nameForCondition()
+												)
+											: module.nameForCondition()
```

Comment:
We use never use `path.relative` here, we use built-in cachable path converted, look at `makeRelative`